search/elastic.py

```diff
@@ -135,6 +135,9 @@ def _process_exclude_dictionary(exclude_dictionary):
             exclude_values = [exclude_values]
         not_properties.extend([{"term": {exclude_property: exclude_value}} for exclude_value in exclude_values])
 
+    if not not_properties:
```

Comment:
Can you add some more information/context in the docstring and or comments around what 'not_properties' represents, how it's used, expected scenarios, what it means to return empty set, etc.